### ISSUE TITLE
Envoy config runtime update

### ISSUE THREAD

**Issue by DimitarHKostov** [envoyproxy/envoy#28833]
*Envoy config runtime update*

*We wonder what is the best way to update Envoy configuration using XDS at runtime and route requests to the newly added configurations seammlessly. Currently we have a route to cluster where the host and port point to the same listener where the request firstly came from. This way we simulate the original request by envoy calling itself. We do this because in the context of one request we can only route through routes to clusters which were already defined and if a new route/cluster is added there is no way to use them. Is there another way to do it? Does envoy have native solution handling such scenarios?*

**Comment by htuch** [comment]
Not sure I can fully understand what is being done here or asked for, can you rephrase?

**Comment by KNikovv** [comment]
@htuch 
We have one scenario where we need to add envoy configuration during the http request.
Basically we are adding cluster,route and secret configuration on the same listener during the request.  
After that we are basically checking that the currently added route is added to Envoy by extracting the route configuration from this url:
`"http://localhost:2001/config_dump?resource=dynamic_route_configs&name_regex=%s"`
We make this check until the route is present, and after that the request proceeeds to that route.

It works ok most of the time, but if for example we make parallel requests or the configuration starts to grow a lot - for some of the requests Envoy starts to fail with 404. And this is observed in the logs :
`Sending local reply with details route_not_found
`

Also to note:
We are using Envoy 1.21.4 with Envoy gRPC external processor server and xds server for the configurations

**Comment by htuch** [comment]
Unless you're using one of the on-demand xDS, e.g. VHDS or on-demand CDS/SRDS, where the request waits for a config update, there's no guarantee that any configuration update will influence any in-flight requests (and in many cases it won't).

**Comment by DimitarHKostov** [comment]
Can you give us an example configuration in Golang for the VHDS?
